code/modules/antagonists/heretic/items/heretic_armor.dm

```diff
@@ -92,12 +92,27 @@
 
 /obj/item/clothing/suit/hooded/cultrobes/void/Initialize(mapload)
 	. = ..()
-
 	create_storage(storage_type = /datum/storage/pockets/void_cloak)
+	make_visible()
 
-/obj/item/clothing/suit/hooded/cultrobes/void/Initialize(mapload)
+/obj/item/clothing/suit/hooded/cultrobes/void/equipped(mob/user, slot)
 	. = ..()
-	make_visible()
+	if(slot & ITEM_SLOT_OCLOTHING)
+		RegisterSignal(user, COMSIG_MOB_EQUIPPED_ITEM, PROC_REF(hide_item))
+		RegisterSignal(user, COMSIG_MOB_UNEQUIPPED_ITEM, PROC_REF(show_item))
+
+/obj/item/clothing/suit/hooded/cultrobes/void/dropped(mob/user)
+	. = ..()
+	UnregisterSignal(user, list(COMSIG_MOB_UNEQUIPPED_ITEM, COMSIG_MOB_EQUIPPED_ITEM))
+
+/obj/item/clothing/suit/hooded/cultrobes/void/proc/hide_item(obj/item/item, slot)
+	SIGNAL_HANDLER
+	if(slot & ITEM_SLOT_SUITSTORE)
+		ADD_TRAIT(item, TRAIT_NO_STRIP, REF(src)) // i'd use examine hide but its a flag and yeah
```

Comment:
"and yeah" what, why not use `EXAMINE_SKIP`? not that I think this is wrong, just wondering

Reply:
Because the item might have that flag already and I have no way to check if the flag is sourced from this item, which means that I might end up removing the flag from an item that is meant to always have it

Reply:
You can always just check `initial(item_flags)` to see if it has the flag by default or not